src/main/java/uk/nhs/hee/tis/revalidation/connection/service/ElasticSearchService.java

```diff
@@ -18,22 +17,83 @@
   private ExceptionElasticSearchRepository exceptionElasticSearchRepository;
 
   /**
-   * add new exceptions to elasticsearch index
+   * add new exceptions to elasticsearch index.
    *
    * @param dataToSave exceptions to go in elasticsearch
    */
-  public void addExceptionViews(List<ExceptionView> dataToSave) {
-    if (CollectionUtils.isNotEmpty(dataToSave)) {
-      exceptionElasticSearchRepository.saveAll(dataToSave);
+  public void saveExceptionViews(ExceptionView dataToSave) {
+    Iterable<ExceptionView> existingRecords = findExceptionViewsByGmcNumberPersonId(dataToSave);
+
+    // if trainee already exists in ES index, then update the existing record
+    if (Iterables.size(existingRecords) > 0) {
+      updateExceptionViews(existingRecords, dataToSave);
+    }
+    // otherwise, add a new record
+    else {
+      addExceptionViews(dataToSave);
     }
   }
 
   /**
-   * remove exceptions from elasticsearch index
+   * remove exceptions from elasticsearch index by gmcReferenceNumber.
    *
    * @param gmcReferenceNumber id of exception to remove
    */
-  public void removeExceptionView(String gmcReferenceNumber) {
-    exceptionElasticSearchRepository.deleteById(gmcReferenceNumber);
+  public void removeExceptionViewByGmcNumber(String gmcReferenceNumber) {
+    exceptionElasticSearchRepository.deleteByGmcReferenceNumber(gmcReferenceNumber);
```

Comment:
Do you think we need null checking or try/catch in the "pass through" service methods? (removeExceptionViewByGmcNumber and removeExceptionViewByTcsPersonId)

Reply:
There were null checks in the updateElasticSearchIndex method. But it is a good shout that the null check should be placed in the same method together with the remove code.
Code is updated and pushed again. Thanks